### PR TITLE
spring-boot-security-jwt-1

### PR DESCRIPTION
**Pull request by DojiDoe**

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/security/AuthenticationService.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class AuthenticationService {
+    private final JwtUtil jwtUtil;
+
```
Comment (JJJazl) [inline src/main/java/mate/academy/bookstore/security/AuthenticationService.java]:
```suggestion
```

File: src/main/java/mate/academy/bookstore/security/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
+            return bearerToken.substring(7);
+        }
```
Comment (JJJazl) [inline src/main/java/mate/academy/bookstore/security/JwtAuthenticationFilter.java]: Move these values to the constants

File: src/main/java/mate/academy/bookstore/security/JwtUtil.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+@Component
+public class JwtUtil {
+    private Key secret;
+
+    @Value("${jwt.expiration}")
+    private long expiration;
+
+    public JwtUtil(@Value("${jwt.secret}") String secretString) {
+        secret = Keys.hmacShaKeyFor(secretString.getBytes(StandardCharsets.UTF_8));
+    }
```
Comment (JJJazl) [inline src/main/java/mate/academy/bookstore/security/JwtUtil.java]:
```suggestion
    private final long expiration;

    public JwtUtil(@Value("${jwt.secret}") String secretString, @Value("${jwt.expiration}") long expiration) {
        this.secret = Keys.hmacShaKeyFor(secretString.getBytes(StandardCharsets.UTF_8));
        this.expiration = expiration;
    }
```

**Review by JJJazl — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/bookstore/controller/AuthenticationController.java [hunk 4]
```diff
@@ -18,10 +21,16 @@
...
+    private final AuthenticationService authenticationService;
 
     @PostMapping("/registration")
     @Operation(summary = "register User")
     public UserResponseDto registerUser(@RequestBody @Valid UserRegistrationRequestDto requestDto) {
         return userService.register(requestDto);
     }
+
+    @PostMapping("/login")
+    public UserLoginResponseDto login(@RequestBody UserLoginRequestDto requestDto) {
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/bookstore/controller/AuthenticationController.java]: annotation Valid missed

File: src/main/java/mate/academy/bookstore/security/JwtAuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            Authentication authentication = new UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader(AUTHORIZATION);
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/bookstore/security/JwtAuthenticationFilter.java]:
```suggestion
        String bearerToken = request.getHeader(HttpHeader.AUTHORIZATION);
```

File: src/main/java/mate/academy/bookstore/security/JwtUtil.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+import java.security.Key;
+import java.util.Date;
+import java.util.function.Function;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class JwtUtil {
+    private Key secret;
+
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/bookstore/security/JwtUtil.java]:
```suggestion
```

File: src/main/java/mate/academy/bookstore/security/JwtUtil.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+import java.nio.charset.StandardCharsets;
+import java.security.Key;
+import java.util.Date;
+import java.util.function.Function;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class JwtUtil {
+    private Key secret;
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/bookstore/security/JwtUtil.java]:
```suggestion
    private final SecretKey secret;
```

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s fix a few minor comments ;)

File: src/main/java/mate/academy/bookstore/controller/AuthenticationController.java [hunk 8]
```diff
@@ -18,10 +21,16 @@
...
     private final UserService userService;
+    private final AuthenticationService authenticationService;
 
     @PostMapping("/registration")
     @Operation(summary = "register User")
     public UserResponseDto registerUser(@RequestBody @Valid UserRegistrationRequestDto requestDto) {
         return userService.register(requestDto);
     }
+
+    @PostMapping("/login")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/controller/AuthenticationController.java:32]: add swagger annotations to controller

File: src/main/java/mate/academy/bookstore/security/HttpHeader.java [hunk 9]
```diff
@@ -0,0 +1,5 @@
+package mate.academy.bookstore.security;
+
+public class HttpHeader {
+    public static final String AUTHORIZATION = "Authorization";
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/security/HttpHeader.java]: remove, you can use Spring constant

File: src/main/java/mate/academy/bookstore/security/JwtAuthenticationFilter.java [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+    public static final String BEARER = "Bearer ";
+    public static final int BEGIN_INDEX = 7;
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/security/JwtAuthenticationFilter.java]:
```suggestion
```

File: src/main/java/mate/academy/bookstore/security/JwtAuthenticationFilter.java [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+            Authentication authentication = new UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader(HttpHeader.AUTHORIZATION);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/security/JwtAuthenticationFilter.java]: import from `import org.springframework.http.HttpHeaders;`

```suggestion
        String bearerToken = request.getHeader(HttpHeaders.AUTHORIZATION);
```

File: src/main/java/mate/academy/bookstore/security/JwtAuthenticationFilter.java [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith(BEARER)) {
+            return bearerToken.substring(BEGIN_INDEX);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/security/JwtAuthenticationFilter.java]:
```suggestion
            return bearerToken.substring(BEARER.length());
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**